### ISSUE TITLE
False-positive E1101 (no-member) inheriting from `set[int]`

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#10031]
### Bug description

```python
# pylint: disable=missing-docstring
class Ints(set[int]):
    def __init__(self):
        self |= {0}
        self.member = True

    def method(self) -> None:
        print(self.member)

print(list(Ints()))
print(Ints().member)
```

### Command used

pylint bug.py

### Pylint output

```
************* Module bug
bug.py:8:14: E1101: Instance of 'Ints' has no 'member' member (no-member)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 2.86/10, +0.00)
```

### Expected behavior

No error.

### Pylint version

pylint 3.3.1
astroid 3.3.5
Python 3.13.0 (tags/v3.13.0:60403a5, Oct  7 2024, 09:38:07) [MSC v.1941 64 bit (AMD64)]

### OS / Environment

Windows 11 24H2

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce on main.
